AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestExecutionStore.kt

```diff
@@ -0,0 +1,59 @@
+package dev.androidx.ci.testRunner
+
+import dev.androidx.ci.datastore.DatastoreApi
+import dev.androidx.ci.firebase.FirebaseTestLabApi
+import dev.androidx.ci.firebase.ToolsResultApi
+import dev.androidx.ci.gcloud.GoogleCloudApi
+
+internal class TestExecutionStore(
+    private val toolsResultApi: ToolsResultApi,
+    private val googleCloudApi: GoogleCloudApi,
+    firebaseProjectId: String,
+    datastoreApi: DatastoreApi,
+    firebaseTestLabApi: FirebaseTestLabApi,
+    gcsResultPath: String
+) {
+    private val testMatrixStore = TestMatrixStore(
+        firebaseProjectId = firebaseProjectId,
+        datastoreApi = datastoreApi,
+        firebaseTestLabApi = firebaseTestLabApi,
+        toolsResultApi = toolsResultApi,
+        resultsGcsPrefix = googleCloudApi.getGcsPath("aosp-ftl/$gcsResultPath")
+    )
+    private val testLabController = FirebaseTestLabController(
+        firebaseTestLabApi = firebaseTestLabApi,
+        firebaseProjectId = firebaseProjectId,
+        testMatrixStore = testMatrixStore
+    )
+
+    private suspend fun getTestExecutionStepsLogcats(
+        projectId: String,
+        historyId: String,
+        executionId: String
+    ): Map<String, String> {
+        val steps = toolsResultApi.listSteps(projectId, historyId, executionId).steps
+        val output = HashMap<String, String>()
+        steps?.forEach { step ->
+            step.testExecutionStep?.toolExecution?.toolOutputs?.forEach {
+                if (it.testCase != null) {
+                    it.output?.fileUri?.let { it1 -> output.put(it1, it.testCase.className + '_' + it.testCase.name) }
+                }
+            }
+        }
+        return output
+    }
+
+    suspend fun getTestExecutionStepsLogcats(
```

Comment:
we probably want to return a structured data here (instead of a hashmap, actual class).
also, we likely want to ask them to pass in FTL TestMatrix instance such that we can grab the projectId/historyId from there and query using it.